hw_8_context_manager.py

```diff
@@ -0,0 +1,175 @@
+import os
+from contextlib import contextmanager
+from contextlib import suppress
+from time import time
+from contextlib import ContextDecorator
+from functools import wraps
+
+existing_folder = 'test1'
+non_existing_folder = 'test2'
+cur_dir = os.getcwd()
+
+# Задача-1
+# Создать объект менеджера контекста который будет переходить в папку которую он принимает на вход.
+# Так же ваш объект должен принимать исключение которое он будет подавлять
+# Если флаг об исключении отсутствует, исключение должно быть поднято.
+
+
+class ContextFolderNavigator:
+    """Custom context manager for opening specified folder.
+    exc_sup parameter:
+    Passed values(e.g., FileNotFoundError) will mean that such exception will be suppressed if  folder doesn't exist.
+    None value is used as default. In this case FileNotFoundError is not suppressed"""
+    def __init__(self, folder, *exc_sup):
+        self._folder = folder
+        self._exc = exc_sup
+
+    def __enter__(self):
+        print('Current folder is ', os.getcwd())
+        try:
+            os.chdir(self._folder)
+        except self._exc:
+            print(f'{self._exc} handled successfully')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        print(f"Now you're in {os.getcwd()}\n")
+
+
+print('\nTask 1. Changing folder with using context manager\n')
+with ContextFolderNavigator(existing_folder):
+    print(f'Running for existing folder')
```

Comment:
Seems like you missed {}. I think, it is supposed to be: `print(f'Running for existing folder {existing_folder}')`